File: javascript/smartlist/smartlist-api-mysql-project/src/modules/grading/dto/response/assessment-question-score.resp.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty } from 'class-validator';
import { CandidateResponseScore } from 'src/entities/candidate-response-score.entity';
export class AssessmentQuestionScoreRespDto {
  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  candidate_id: number;

  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  assessment_id: number;

  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  assessment_block_id: number;

  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  question_id: number;

  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  score: number;

  constructor(candidateResponseScore: CandidateResponseScore) {
    this.candidate_id = candidateResponseScore.candidate.id;
    this.assessment_id = candidateResponseScore.assessment.id;
    this.assessment_block_id = candidateResponseScore.assessmentBlock.id;
    this.question_id = candidateResponseScore.question.id;
    this.score = candidateResponseScore.score;
  }
}

File: java/smartlist/src/main/java/com/smartlist/module/grading/dto/response/AssessmentQuestionScoreRespDTO.java
package com.smartlist.module.grading.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.smartlist.model.CandidateResponseScore;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AssessmentQuestionScoreRespDTO {

    @Schema(description = "Candidate Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "candidate_id")
    Integer candidateId;

    @Schema(description = "Assessment Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "assessment_id")
    Integer assessmentId;

    @Schema(description = "Assessment Block Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "assessment_block_id")
    Integer assessmentBlockId;

    @Schema(description = "Question Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "question_id")
    Integer questionId;

    @Schema(description = "Score", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    Integer score;

}
